docs/src/main/asciidoc/kafka.adoc

```diff
@@ -1548,6 +1548,368 @@ kafka.bootstrap.servers=...
+
+public class FruitDeserializer extends ObjectMapperDeserializer<Fruit> {
+    public FruitDeserializer() {
+        super(Fruit.class);
+    }
+}
+----
+
+The associated configuration would be:
+
+[source, properties]
+----
+mp.messaging.incoming.fruits.connector=smallrye-kafka
+mp.messaging.incoming.fruits.value.deserializer=org.acme.FruitDeserializer
+----
+
+Check <<jackson-serialization>> for more detail about the usage of Jackson with Kafka.
+You can also use Avro.
+
+[#persisting-kafka-messages-with-hibernate-reactive]
+=== Persisting Kafka messages with Hibernate Reactive
+
+To persist objects received from Kafka into a database, you can use Hibernate Reactive with Panache.
+
+Let's imagine you receive `Fruit` objects.
+For simplicity purposes, our `Fruit` class is pretty simple:
+
+[source, java]
+----
+package org.acme;
+
+import javax.persistence.Entity;
+
+import io.quarkus.hibernate.reactive.panache.PanacheEntity;  // <1>
+
+@Entity
+public class Fruit extends PanacheEntity {
+
+    public String name;
+
+}
+----
+<1> Make sure to use the reactive variant
+
+To consume `Fruit` instances stored on a Kafka topic, and persist them into a database, you can use the following approach:
+
+[source, java]
+----
+package org.acme;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+
+import io.quarkus.hibernate.reactive.panache.Panache;
+import io.smallrye.mutiny.Uni;
+
+@ApplicationScoped
+public class FruitStore {
+
+    @Incoming("fruits")
+    public Uni<Void> persist(Fruit fruit) {
+        return Panache.withTransaction(() ->  // <1>
+            fruit.persist()                   // <2>
+                .map(persisted -> null)       // <3>
+        );
+    }
+
+}
+----
+<1> Instruct Panache to run the given (asynchronous) action in a transaction. The transaction completes when the action completes.
+<2> Persist the entity. It returns a `Uni<Fruit>`.
+<3> Switch back to a `Uni<Void>`.
+
+Unlike with _classic_ Hibernate, you can't use `@Transactional`.
+Instead, we use `Panache.withTransaction` and persist our entity.
+The `map` is used to return a `Uni<Void>` and not a `Uni<Fruit>`.
+
+You need a deserializer that can create a `Fruit` from the record.
+This can be done using a Jackson deserializer:
+
+[source, java]
+----
+package org.acme;
+
+import io.quarkus.kafka.client.serialization.ObjectMapperDeserializer;
+
+public class FruitDeserializer extends ObjectMapperDeserializer<Fruit> {
+    public FruitDeserializer() {
+        super(Fruit.class);
+    }
+}
+----
+
+The associated configuration would be:
+
+[source, properties]
+----
+mp.messaging.incoming.fruits.connector=smallrye-kafka
+mp.messaging.incoming.fruits.value.deserializer=org.acme.FruitDeserializer
+----
+
+Check <<jackson-serialization>> for more detail about the usage of Jackson with Kafka.
+You can also use Avro.
+
+=== Writing entities managed by Hibernate to Kafka
```

Comment:
Do we want a Hibernate Reactive variant of this section?

Reply:
Added.